aws/resource_aws_cloudwatch_dashboard.go

```diff
@@ -0,0 +1,139 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCloudWatchDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceAwsCloudWatchDashboardCreate,
+		Read:          resourceAwsCloudWatchDashboardRead,
+		Update:        resourceAwsCloudWatchDashboardUpdate,
+		Delete:        resourceAwsCloudWatchDashboardDelete,
+		SchemaVersion: 1,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"dashboard_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dashboard_body": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateJsonString,
+				StateFunc: func(v interface{}) string {
+					json, _ := normalizeJsonString(v)
+					return json
+				},
+			},
+			"dashboard_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudWatchDashboardCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatchconn
+
+	params := getAwsCloudWatchPutDashboardInput(d)
+
+	log.Printf("[DEBUG] Creating CloudWatch Dashboard: %#v", params)
+	_, err := conn.PutDashboard(&params)
+	if err != nil {
+		return fmt.Errorf("Creating dashboard failed: %s", err)
+	}
+	d.SetId(d.Get("dashboard_name").(string))
+	log.Println("[INFO] CloudWatch Dashboard created")
+
+	return resourceAwsCloudWatchDashboardRead(d, meta)
+}
+
+func resourceAwsCloudWatchDashboardRead(d *schema.ResourceData, meta interface{}) error {
+	a, err := getAwsCloudWatchDashboard(d, meta)
+	if err != nil {
+		return err
+	}
+	if a == nil {
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("[DEBUG] Reading CloudWatch Dashboard: %s", d.Get("dashboard_name"))
+
+	d.Set("dashboard_name", a.DashboardName)
+	d.Set("dashboard_body", a.DashboardBody)
+	return nil
+}
+
+func resourceAwsCloudWatchDashboardUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatchconn
+	params := getAwsCloudWatchPutDashboardInput(d)
+
+	log.Printf("[DEBUG] Updating CloudWatch Dashboard: %#v", params)
+	_, err := conn.PutDashboard(&params)
+	if err != nil {
+		return fmt.Errorf("Updating dashboard failed: %s", err)
+	}
+	log.Println("[INFO] CloudWatch Dashboard updated")
+
+	return resourceAwsCloudWatchDashboardRead(d, meta)
+}
+
+func resourceAwsCloudWatchDashboardDelete(d *schema.ResourceData, meta interface{}) error {
+	p, err := getAwsCloudWatchDashboard(d, meta)
```

Comment:
It would be better to avoid this call, and check the DeleteDashboards function return codes for the `ResourceNotFound` awserr.
Thus, we could move the code from `getAwsCloudWatchDashboard` to the read function only